src/main/java/de/tum/in/www1/artemis/repository/specs/PostSpecs.java

```diff
@@ -109,15 +109,27 @@ public static Specification<Post> getOwnSpecification(boolean filterToOwn, Long
      * @param userId            id of the calling user
      * @return specification used to chain DB operations
      */
-    public static Specification<Post> getAnsweredOrReactedSpecification(boolean answeredOrReacted, Long userId) {
+    public static Specification<Post> getAnsweredOrReactedSpecification(boolean answeredOrReacted, Long userId, PostSortCriterion postSortCriterion, SortingOrder sortingOrder) {
         return ((root, query, criteriaBuilder) -> {
-            if (!answeredOrReacted) {
-                return null;
-            }
-            else {
-                Join<Post, AnswerPost> joinedAnswers = root.join(Post_.ANSWERS, JoinType.LEFT);
+            Join<Post, AnswerPost> joinedAnswers = null;
+
```

Comment:
Can't you return early? At least for the part with `if (answeredOrReacted)`.

Reply:
Not really, since answeredOrReacted and sorting by number of answers can be independently set. I am doing both in one method, since I only want to join with answers one. I added a comment.

Reply:
I reverted the changes to this method